### ISSUE TITLE
Upload package in AUR

### ISSUE THREAD

**Issue by ghost** [throneproj/Throne#163]
## Description suggestions

Please Release Package in [AUR](https://aur.archlinux.org) for Arch Users.

**Comment by Mahdi-zarei** [comment]
I need to change the name of the app for this, since Nekoray is already present there from the original repo.

**Comment by Turbine8845** [comment]
> I need to change the name of the app for this, since Nekoray is already present there from the original repo.

Can you please create a tar.zst package for Arch users? This would allow Nekoray to be installed and managed using pacman. You can refer to the following page in the Arch Wiki for guidance:
https://wiki.archlinux.org/title/Creating_packages

**Comment by lufog** [comment]
Added [PKGBUILD (bin ver)](https://aur.archlinux.org/packages/nekoray-mahdi-zarei-bin) to AUR. 

@Mahdi-zarei, if you decide to rename the app and/or create an official package. Tag me here (or there), and I will remove (or disown) the package.

**Comment by Turbine8845** [comment]
> Added [PKGBUILD (bin ver)](https://aur.archlinux.org/packages/nekoray-mahdi-zarei-bin) to AUR.
> 
> [@Mahdi-zarei](https://github.com/Mahdi-zarei), if you decide to rename the app and/or create an official package. Tag me here (or there), and I will remove (or disown) the package.

Thank you for adding this version of nekoray to AUR. However, why didn't you configured the PKGBUILD to build Mahdi's nekoray from the source?

**Comment by lufog** [comment]
@Turbine8845, it doesn't matter to me whether to build from source or use binaries. Considering that the latter is much faster and does not require installing build dependencies, I chose the latter. And I doubt that, in this particular case, the user will see any difference, even with `-march=native` or similar optimizations, because of which people usually prefer building from source.

**Comment by NoPlagiarism** [comment]
> I need to change the name of the app for this

It's been a long time, and now original repo is archived. Any thoughts about rebranding/renaming or sth?

**Comment by Mahdi-zarei** [comment]
> > I need to change the name of the app for this
> 
> It's been a long time, and now original repo is archived. Any thoughts about rebranding/renaming or sth?

Yes I think it's about time to do it. I'm also open to name suggestions, for the time being I'm considering the name `Throne` for it, partly influenced by WC3.

**Comment by lufog** [comment]
I have nothing against a completely new name, I'll just suggest an alternative in case if you want to keep existing discoverability, you can simply add `-mz` to the old name (although this is the most boring solution). Similar renaming of forks come to mind:
`proton` -> `proton-ge`
`mkxp` -> `mkxp-z`
`nestopia` -> `nestopia-ue`

**Comment by Mahdi-zarei** [comment]
Since there were no other suggestions, I guess I will go on with the name `Throne`. I will keep nekoray in about and readme for better discoverability as per @lufog's suggestion. There are lots of new things in sing-box 1.12, and there are some major refactorings that need to be done, so I guess I will rebrand the project once those changes have been made and we have a stable and up to date code at hand. Will also try to see if we can add the project to snap as well as work on the AUR package with the help of @lufog and get the pkg-build going as well.

**Comment by lufog** [comment]
@Mahdi-zarei, prepared a PKGBUILD, with a new name, and assumed (`throne`, `throne_core`, `throne.png`) file names. So after the rebrand/rename, poke me, and I'll update the AUR page (after a little local check that I didn't mess it up).

**Comment by NoPlagiarism** [comment]
Bumping this, since there already been some mix-ups with [nekoray-bin.git](https://aur.archlinux.org/cgit/aur.git/log/?h=nekoray-bin). It was moved to the "Throne" and then reversed to Matsuri original. 

P.S: I know about [nekoray-mahdi-zarei-bin](https://aur.archlinux.org/packages/nekoray-mahdi-zarei-bin). Still renaming should be done, since this kind of fuck-up could happen again

**Comment by lufog** [comment]
@Mahdi-zarei, I recently saw a mention that a renaming is coming, and that with it the version will change to 1.0.0. Maybe it makes sense to exclude the date with the new numbering? To me, it just adds clutter, being an additional factor that has to be monitored/updated in the packaging script.

**Comment by Mahdi-zarei** [comment]
Yes i guess we will drop this. The versioning should become like `v1.2.3` and for pre-releases `v1.2.3-beta.5` etc.

**Comment by Mahdi-zarei** [comment]
The code refactor for rebranding is almost done, I would appreciate if you could submit a PR or instruct on how to create and maintain an AUR package for Throne.

**Comment by lufog** [comment]
@Mahdi-zarei, I propose to do it this way, I continue to maintain (updated) PKGBUILD bin[1] version for Arch. Maintenance of source[2] and git[3] versions will be continued by the maintainers of source and git versions of Nekoray, since they are already using this fork.

And in the readme you can simply add a link to the Throne(-git/-bin) pages in AUR. You don't have to worry about one of the maintainers disappearing. AUR is made in such a way that if one maintainer disappears, another can take over the build script maintenance.

To update my PKGBUILD, I only need an archive with the compiled program, in the form in which it will be present in releases.

---
1. Bin version, the package is built from pre-compiled binaries downloaded during the build process from the official site/page of the program.
2. Source version, builds the program from the source, then builds a package from the resulting binaries. Only this version (with rare exceptions) [can get into the official Arch repository](https://wiki.archlinux.org/title/Arch_User_Repository#How_to_get_a_PKGBUILD_into_the_extra_repository?).
3. Git version, builds the program from the source (latest commits from git), then builds a package from the resulting binaries.

**Comment by Mahdi-zarei** [comment]
Seems great. Send me the links so that I can include them in the readme. Also if possible, notify the maintainers of the other versions so that they could rename and remove ambiguity.

**Comment by lufog** [comment]
@Mahdi-zarei, I informed the active maintainer of the other two PKGBUILDs about the renaming. And regarding new pages, they (not counting the -git version) can only be created after the release (with a new name). Since it would be wrong to upload build scripts that (for now) have nothing to build from (tagged release/prebuild binaries).

**Comment by lufog** [comment]
@Mahdi-zarei, given the predictability of links/naming in the AUR, the links will be [throne](https://aur.archlinux.org/packages/throne)([-git](https://aur.archlinux.org/packages/throne-git)/[-bin](https://aur.archlinux.org/packages/throne-bin)).

**Comment by Mahdi-zarei** [comment]
Great! There a couple of stuff to sort out, and then i guess we can have the release.

**Comment by lufog** [comment]
@Mahdi-zarei, by the way, I just noticed that executable/folder names now start with a capital letter. I would recommend leaving these names with a lowercase letter. This will help avoid problems on case-sensitive file systems, where `core` and `Core` are two different files that can be in the same folder.

**Comment by Mahdi-zarei** [comment]
Ok I will refactor them later.

**Comment by lufog** [comment]
Well, there is some problem with renaming `Core` to `core` on linux.

<details>

<summary>Before rename</summary>

<img width="530" height="799" alt="Image" src="https://github.com/user-attachments/assets/bec1c2b2-f3c3-409e-90d4-35845e1c4bda" />

</details>


<details>

<summary>After rename</summary>

<img width="530" height="744" alt="Image" src="https://github.com/user-attachments/assets/f03ea44a-f757-436e-8481-806431c2c9f6" />

</details>

I'm not sure if this is a file manager bug or `core` is a reserved name.

**Comment by Mahdi-zarei** [comment]
Oh, this makes things complicated, maybe we stay with throne and core both being capital and others as all lower case?

**Comment by Mahdi-zarei** [comment]
The reason for it being recognised as a crash is most likely because linux crash files are named core dumps, maybe they are saved as simply core.

**Comment by lufog** [comment]
> Oh, this makes things complicated, maybe we stay with throne and core both being capital and others as all lower case?

Yes, I think this is the most reasonable solution, especially since for convenience the program is launched through a wrapper script `/usr/bin/throne` (at least on Arch), which can be named with a lowercase letter.

```bash
#!/bin/sh

/usr/lib/throne/Throne -- -appdata
```

**Comment by Mahdi-zarei** [comment]
I have released the beta version.

**Comment by lufog** [comment]
@Mahdi-zarei, Should I use [this file](https://github.com/throneproj/Throne/blob/dev/res/public/Throne.png) as an icon?

Also, I noticed a regression, now after restarting the program, the sorting by columns is reset, before program rename it was saved.

**Comment by lufog** [comment]
The new `PKGBUILD` is ready, anyone wishing to test it can install [throne-bin](https://aur.archlinux.org/packages/throne-bin).
